### ISSUE TITLE
[Bug]: SDPA fwd ragged Stats (LSE) written only for the first head of each kv group when s_q == 1 with GQA (cuDNN 9.26/9.27, all archs)

### ISSUE THREAD

**Issue by YangXu1990uiuc** [NVIDIA/cudnn-frontend#1083]
### Summary
Unified SDPA forward, THD/ragged layout, `generate_stats=True` with a **ragged-offset Stats tensor**, `s_q == 1` and GQA (`h_q > h_kv`): the Stats (LSE) is written only for the **first q head of every kv group**; the other `h_q/h_kv - 1` heads' entries are never written (whatever was in the buffer survives). `O` is correct. Padded (non-ragged) Stats, MHA, and `s_q >= 2` are all correct.

### Environment
cuDNN 9.26.0.51 and 9.27.0.21 (torch-bundled / NVIDIA PyPI), cudnn-frontend 1.29 / 1.30. Reproduces on SM80 (A100), SM90 (H100) and SM100 (B200-class), so it is not arch-specific; the kernel that runs for this shape class is the `fort_native_sdpa_sm80_flash_fprop_wmma_*` family on every arch.

### Repro (python frontend)
```python
# b=1, h_q=8, h_kv=2, d=128, kv 300 tokens, q 1 token, bf16
q = gr.tensor(dim=(b, H, 1, D), stride=(H*D, D, H*D, 1)); q.set_ragged_offset(rq); q.set_ragged_offset_multiplier(H*D)
k/v likewise (ragged, multiplier HKV*D); o likewise
o, st = gr.sdpa(q=q, k=k, v=v, attn_scale=1/sqrt(D), use_padding_mask=True, cu_seq_len_q=cq, cu_seq_len_kv=ck, generate_stats=True)
st.set_output(True).set_dim((b, H, 1, 1)).set_stride((H, 1, H, 1)).set_data_type(FLOAT)
st.set_ragged_offset(rs); st.set_ragged_offset_multiplier(H)   # packed [tokens, H] LSE
```
Fill the LSE buffer with NaN before execute: only slots 0 and 4 (heads 0 and 4 = the first head of each kv group) are written. 32/8 heads: 24 of 32 wrong per token. Compare with `torch.logsumexp` of the scaled scores: the written heads match to <1e-3.

### Root cause (found)
In the runtime-compiled engine's global-store emitter, the ragged ("packed stats") ROW_REDUCE pointer decl steps the row term by `strides[2]` (the sequence stride) unconditionally, while the padded-Stats branch switches to `strides[1]` (head stride) in the decode class (`s_q == 1`, where the tile rows are the q heads of a kv group). The kernel tensor descriptor zeroes the stride of every size-1 dim, and dim 2 is 1 in that class, so every row of the tile resolves to the group's first slot. One-line fix; tracked internally as **NVBug 6783545**, fix MR in review.

### Impact / workaround
FlashInfer's `backend="auto"` ragged prefill routes to cuDNN on SM100 and vLLM's decode-context-parallel prefill merges by LSE, so a step whose requests all contribute one token silently merges with a wrong LSE for 3/4 of the heads. FlashInfer keeps `max_q_len == 1` GQA steps off cuDNN until a release carries the fix (flashinfer-ai/flashinfer#5245). Declaring the Stats tensor **padded** (`[b, h, 1, 1]`, no ragged offset) is a correct workaround.

<sub>note to self: claude::61d24ed2-7c90-4a97-9cbb-b91ae18136fd — "flashinfer frost prefill/GEMM enablement" · cwd /home/scratch.yanxu_libs/flashinfer</sub>
